animation_nodes/nodes/geometry/intersect_line_line.py

```diff
@@ -1,36 +1,81 @@
-        if "closest2" in required: yield "    closest2 = Vector((0, 0, 0))"
-        if "isValid" in required:  yield "    isValid = False"
-        yield "else:"
-        if "closest1" in required: yield "    closest1 = closestPoints[0]"
-        if "closest2" in required: yield "    closest2 = closestPoints[1]"
-        if "isValid" in required:  yield "    isValid = True"
-
-    def getUsedModules(self):
-        return ["mathutils"]
+        self.newInput(VectorizedSocket("Vector", "useFirstLineStartList",
+            ("First Line Start", "firstLineStart", dict(value = (1, 1, 0))),
+            ("First Line Starts", "firstLineStarts"),
+            codeProperties = dict(default = (1, 1, 0))))
+        self.newInput(VectorizedSocket("Vector", "useFirstLineEndList",
+            ("First Line End", "firstLineEnd", dict(value = (-1, -1, 0))),
+            ("First Line Ends", "firstLineEnds"),
+            codeProperties = dict(default = (-1, -1, 0))))
+
+        self.newInput(VectorizedSocket("Vector", "useSecondLineStartList",
+            ("Second Line Start", "secondLineStart", dict(value = (1, -1, 0))),
+            ("Second Line Starts", "secondLineStarts"),
+            codeProperties = dict(default = (1, -1, 0))))
+        self.newInput(VectorizedSocket("Vector", "useSecondLineEndList",
+            ("Second Line End", "secondLineEnd", dict(value = (-1, 1, 0))),
+            ("Second Line Ends", "secondLineEnds"),
+            codeProperties = dict(default = (-1, 1, 0))))
+
+        props = ["useFirstLineStartList", "useFirstLineEndList",
+        "useSecondLineStartList", "useSecondLineEndList"]
+
+        self.newOutput(VectorizedSocket("Vector", props,
+            ("First Nearest Point", "firstNearestPoint"),
+            ("First Nearest Points", "firstNearestPoints")))
+        self.newOutput(VectorizedSocket("Vector", props,
+            ("Second Nearest Point", "secondNearestPoint"),
+            ("Second Nearest Points", "secondNearestPoints")))
+
+        self.newOutput(VectorizedSocket("Float", props,
+            ("First Parameter", "firstParameter"),
+            ("First Parameters", "firstParameters")))
+        self.newOutput(VectorizedSocket("Float", props,
+            ("Second Parameter", "secondParameter"),
+            ("Second Parameters", "secondParameters")))
+
+        self.newOutput(VectorizedSocket("Boolean", props,
+            ("Valid", "valid"),
+            ("Valids", "valids")))
+
+    def getExecutionFunctionName(self):
+        useList = any((self.useFirstLineStartList, self.useFirstLineEndList,
+        self.useSecondLineStartList, self.useSecondLineEndList))
+        if useList:
+            return "execute_List"
+        else:
+            return "execute_Single"
+
+    def execute_List(self, firstLineStarts, firstLineEnds, secondLineStarts, secondLineEnds):
+        firstLineStarts = VirtualVector3DList.fromListOrElement(firstLineStarts,
```

Comment:
Don't break the line here.
It is likely that I will rename the `fromListOrElement` function soon to give it a shorter name.
Also, almost always when you do a line break your should indent the lower line to make it easier to see that it belongs to the line before.